src/pyspiffe/svid/x509_svid.py

```diff
@@ -1,34 +1,475 @@
 
         Args:
-            spiffe_id: SPIFFE ID of the JWT-SVID as present in the 'sub' claim.
-            audience: Audience is the intended recipients of JWT-SVID as present in the 'aud' claim.
-            expiry: Expiration time of JWT-SVID as present in 'exp' claim.
-            claims: Parsed claims from token.
-            token: Serialized JWT token.
+            x509_svid: the 'X509Svid' that has the certs_chain and private_key to be saved on disk.
+
+            certs_chain_path (str): path to the file containing one or more X.509 certificates as PEM blocks.
+                                    The certs_chain file is configured with a filemode = '0644'.
+
+            private_key_path (str): path the file containing a PKCS#8 PEM block.
+                                    The private_key file is configured with a filemode = '0600'.
+
+            encoding (serialization.Encoding): the encoding used to serialize the certs and private key, can be
+                                            serialization.Encoding.PEM or serialization.Encoding.DER.
+
+        Raises:
+            ValueError: in case the encoding is not either PEM or DER (from serialization.Encoding).
+
+            X509SvidError: in case the certs chain or the private key in the X509Svid cannot be converted to bytes.
+
+            StorePrivateKeyError: in the case there is an error storing the private key to the file.
+
+            StoreCertificateError: in the case the file path in certs_chain_path cannot be open to write,
+                                  or there is an error storing the certificates to the file.
         """
-        self.spiffe_id = spiffe_id
-        self.audience = audience
-        self.expiry = expiry
-        self.claims = claims
-        self.token = token
+
+        if not (encoding is encoding.PEM or encoding is encoding.DER):
+            raise ValueError(
+                'Encoding not supported: {}. Expected \'PEM\' or \'DER\'.'.format(
+                    encoding
+                )
+            )
+
+        cls._write_certs_to_file(certs_chain_path, encoding, x509_svid)
+
+        private_key_bytes = cls._extract_private_key_bytes(
+            encoding, x509_svid.private_key()
+        )
+        cls._write_private_key_to_file(private_key_path, private_key_bytes)
+
+    @staticmethod
+    def _load_private_key_bytes(private_key_path: str) -> bytes:
+        try:
+            with open(private_key_path, 'rb') as key_file:
+                return key_file.read()
+        except FileNotFoundError:
+            raise LoadPrivateKeyError(
+                'Private key file not found: {}'.format(private_key_path)
+            )
+        except Exception as err:
+            raise LoadPrivateKeyError(
+                'Private key file could not be read: {}'.format(str(err))
+            )
+
+    @staticmethod
```

Comment:
To me, looks like all static methods here can be moved from the X509Svid object to simple functions in a different module.

Reply:
do you think? These static methods are very specific to the X509Svid type and I don't see any possible use for other modules.

Reply:
ok, I refactored those methods as simple functions in the X509Svid module, prefixing them with an underscore for suggesting that they are internal and not meant to be used outside of the library (as an implementation detail they can change later on without breaking the interface contract).